apis/externalsecrets/v1beta1/secretstore_vault_types.go

```diff
@@ -237,3 +240,20 @@ type VaultCertAuth struct {
 	// authenticate with Vault using the Cert authentication method
 	SecretRef esmeta.SecretKeySelector `json:"secretRef,omitempty"`
 }
+
+// VaultIamAuth authenticates with Vault using the IAM authentication method
+type VaultIamAuth struct {
+
+	// Path where the AWS auth method is enabled in Vault, e.g: "aws"
+	Path string `json:"path"`
+	// AWS region
+	Region string `json:"region"`
+	//Vault Role. In vault, a role describes an identity with a set of permissions, groups, or policies you want to attach a user of the secrets engine
+	Role string `json:"vaultRole"`
+	// Specify credentials in a Secret object
+	// +optional
+	SecretRef *AWSAuthSecretRef `json:"secretRef,omitempty"`
```

Comment:
Isn't this block here effectively coupling AWS and vault providers? Do we want that?

Reply:
Thanks for your review. Yes you are right, this is coupling them. I can decouple them. Let me add a fix for this. Thanks.

Reply:
@gusfcarvalho : I have now decoupled the providers. Please review. Thanks.